File: InvotraxApp/backend/InvotraxApp/src/main/java/dev/lkeleti/invotraxapp/dto/PartnerDto.java
package dev.lkeleti.invotraxapp.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class PartnerDto {
    private Long id;
    private String name;
    private ZipCodeDto zipCode;
    private String streetName;
    private String streetType;
    private String houseNumber;
    private String building;
    private String staircase;
    private String floor;
    private String door;
    private String landRegistryNumber;
    private boolean isPrivate;
    private String taxNumber;
    private String email;
    private String phoneNumber;
    private PaymentMethodDto preferredPaymentMethod;
    private LocalDateTime createdAt;
    private BigDecimal balance;
    private Integer defaultPaymentDeadline;
    private String bankName;
    private String bankNumber;
    private String iban;
    private boolean deleted;
}

File: InvotraxApp/frontend/invotrax/src/app/models/partner.dto.ts
import { ZipCodeDto } from "./zip-code.dto";

export interface PartnerDto {
    id: number;
    name: string;
    zipCode: ZipCodeDto;

    streetName: string;
    streetType: string;
    houseNumber: string;
    building: string;
    staircase: string;
    floor: string;
    door: string;
    landRegistryNumber: string;

    isPrivate: boolean;
    taxNumber: string;
    email: string;
    phoneNumber: string;
    preferredPaymentMethod: string;
    createdAt: Date;
    balance: number;
    defaultPaymentDeadline: number;
    bankName: string;
    bankNumber: string;
    iban: string;
    deleted: boolean;
}